storm-core/src/jvm/org/apache/storm/zookeeper/Zookeeper.java

```diff
@@ -0,0 +1,355 @@
+        return mkClient(conf, servers, port, root, new DefaultWatcherCallBack(), authConf);
+    }
+
+    public static CuratorFramework mkClient(Map conf, List<String> servers, Object port, String root, final WatcherCallBack watcher, Map authConf) {
+        CuratorFramework fk;
+        if (authConf != null) {
+            fk = Utils.newCurator(conf, servers, port, root, new ZookeeperAuthInfo(authConf));
+        } else {
+            fk = Utils.newCurator(conf, servers, port, root);
+        }
+
+        fk.getCuratorListenable().addListener(new CuratorListener() {
+            @Override
+            public void eventReceived(CuratorFramework _fk, CuratorEvent e) throws Exception {
+                if (e.getType().equals(CuratorEventType.WATCHED)) {
+                    WatchedEvent event = e.getWatchedEvent();
+
+                    watcher.execute(event.getState(), event.getType(), event.getPath());
+                }
+
+            }
+        });
+        fk.start();
+        return fk;
+    }
+
+    /**
+     * connect ZK, register Watch/unhandle Watch
+     *
+     * @return
+     */
+    public static CuratorFramework mkClient(Map conf, List<String> servers, Object port, String root, final WatcherCallBack watcher) {
+
+        return mkClient(conf, servers, port, root, watcher, null);
+    }
+
+    public static String createNode(CuratorFramework zk, String path, byte[] data, org.apache.zookeeper.CreateMode mode, List<ACL> acls)
+            throws RuntimeException {
+
+        String ret = null;
+        try {
+            String npath = Utils.normalizePath(path);
+            ret = zk.create().withMode(mode).withACL(acls).forPath(npath, data);
+        } catch (Exception e) {
+            throw Utils.wrapInRuntime(e);
+        }
+        return ret;
+    }
+
+    public static String createNode(CuratorFramework zk, String path, byte[] data, List<ACL> acls) throws RuntimeException {
+        return createNode(zk, path, data, org.apache.zookeeper.CreateMode.PERSISTENT, acls);
+    }
+
+    public static boolean existsNode(CuratorFramework zk, String path, boolean watch) throws RuntimeException {
+        Stat stat = null;
+        try {
+            if (watch) {
+                stat = zk.checkExists().watched().forPath(Utils.normalizePath(path));
+            } else {
+                stat = zk.checkExists().forPath(Utils.normalizePath(path));
+            }
+        } catch (Exception e) {
+            throw Utils.wrapInRuntime(e);
+        }
+
+        return stat != null;
+    }
+
+    public static void deleteNode(CuratorFramework zk, String path) throws RuntimeException {
+        try {
+            String npath = Utils.normalizePath(path);
+            if (existsNode(zk, npath, false)) {
+                zk.delete().deletingChildrenIfNeeded().forPath(Utils.normalizePath(path));
+            }
+
+        } catch (KeeperException.NoNodeException e) {
```

Comment:
In the original code we did a try-cause.  I am not sure if that is needed here or not, I assume not, but I wanted to point it out if we run into issues while testing.